lib/node_modules/@stdlib/math/base/complex/cis/benchmark/c/benchmark.c

```diff
@@ -0,0 +1,123 @@
+/**
+* Benchmark `cis`.
+*/
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <complex.h>
+#include <sys/time.h>
+
+#define NAME "cis"
+#define ITERATIONS 1000000
+#define REPEATS 3
+
+/**
+* Prints the TAP version.
+*/
+void print_version() {
+	printf( "TAP version 13\n" );
+}
+
+/**
+* Prints the TAP summary.
+*
+* @param total     total number of tests
+* @param passing   total number of passing tests
+*/
+void print_summary( int total, int passing ) {
+	printf( "#\n" );
+	printf( "1..%d\n", total ); // TAP plan
+	printf( "# total %d\n", total );
+	printf( "# pass  %d\n", passing );
+	printf( "#\n" );
+	printf( "# ok\n" );
+}
+
+/**
+* Prints benchmarks results.
+*
+* @param elapsed   elapsed time in seconds
+*/
+void print_results( double elapsed ) {
+	double rate = (double)ITERATIONS / elapsed;
+	printf( "  ---\n" );
+	printf( "  iterations: %d\n", ITERATIONS );
+	printf( "  elapsed: %0.9f\n", elapsed );
+	printf( "  rate: %0.9f\n", rate );
+	printf( "  ...\n" );
+}
+
+/**
+* Returns a clock time.
+*
+* @return clock time
+*/
+double tic() {
+	struct timeval now;
+	gettimeofday( &now, NULL );
+	return (double)now.tv_sec + (double)now.tv_usec/1.0e6;
+}
+
+/**
+* Generates a random double on the interval [0,1].
+*
+* @return random double
+*/
+double rand_double() {
+	int r = rand();
+	return (double)r / ( (double)RAND_MAX + 1.0 );
+}
+
+/**
+* Runs a benchmark.
+*
+* @return elapsed time in seconds
+*/
+double benchmark() {
+	double elapsed;
+	double re;
+	double im;
+	double t;
+	int i;
+
+	double complex z1;
+	double complex z2;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		re = ( 100.0*rand_double() ) - 50.0;
+		im = ( 100.0*rand_double() ) - 50.0;
+		z1 = re + im*I;
+
+		z2 = cexp( I * z1 );
```

Comment:
A couple questions:

- `cexp` is not equivalent to `cis`, correct?
- why is the `I * z1` needed when calling `cexp`?

Reply:
That's correct. `cis(x) = cexp(i * x)`. Which is why this module is rather trivial. Though I believe the most common use case is that the argument is pure real and it's just a phasor. That multiplication is because `c` doesn't implement cis. (Obviously `exp` will very very shortly follow this module as a trivial simplification.)

Reply:
From the C docs: http://en.cppreference.com/w/c/numeric/complex/cexp

```
cexp(z) = e^z = e^(x+iy) = e^x cis(y) = e^x (cos(y) + i sin(y))
```
Could you not implement `cis` in C by multiplying through by `1/e^x`?

Reply:
Ah, now I get it. Yes. Doing the algebra first is a better comparison. 👍

Reply:
Switching to:

```
  for ( i = 0; i < ITERATIONS; i++ ) { 
    re = ( 100.0*rand_double() ) - 50.0;
    im = ( 100.0*rand_double() ) - 50.0;

    z2 = ( cos( im ) + I * sin( im ) ) / exp( re );
```

which seems *slightly* faster. It's a little weird since it's not an exact analog, but that doesn't bother me too much since it contains the three fundamental evaluations (sin, cos, exp). The other details are just trivial reorganizations.

Reply:
Sounds fine to me.